test/tests/performance/pbs_preemptperformance.py

```diff
@@ -480,6 +480,69 @@ def test_preemption_basic(self):
             ncpus *= 3
             S_jobs += ncpus
 
+    @timeout(3600)
+    def test_preemption_with_soft_limits(self):
+        """
+        Measure the time scheduler takes to preempt when the high priority
+        job hits soft limits under a considerable amount of workload.
+        """
+        a = {'resources_available.ncpus': 4,
+             'resources_available.mem': '6400mb'}
+        self.server.create_vnodes('vn', a, 500, self.mom, usenatvnode=False)
+        p = "express_queue, normal_jobs, server_softlimits, queue_softlimits"
+        a = {'preempt_prio': p}
+        self.server.manager(MGR_CMD_SET, SCHED, a, runas=ROOT_USER)
+
+        a = {'max_run_res_soft.ncpus': "[u:" + str(TEST_USER)+"=1]"}
+        self.server.manager(MGR_CMD_SET, QUEUE, a, 'workq')
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+
+        # submit a bunch of jobs as TEST_USER2
+        a = {ATTR_l + '.select=1:ncpus': 1}
+        for _ in range(2000):
+            j = Job(TEST_USER2, attrs=a)
+            j.set_sleep_time(3000)
+            self.server.submit(j)
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'True'})
+        self.server.expect(JOB, {'job_state=R': 2000}, interval=10, offset=5,
+                           max_attempts=100)
+
+        qname = 'highp'
+        a = {'queue_type': 'execution', 'priority': '200',
+             'started': 'True', 'enabled': 'True'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, qname)
+
+        self.server.manager(MGR_CMD_SET, SERVER, {'scheduling': 'False'})
+        a = {ATTR_l + '.select=1:ncpus': 1, ATTR_q: 'highp'}
+        jid = ""
+        for _ in range(2000):
+            j = Job(TEST_USER2, attrs=a)
+            j.set_sleep_time(3000)
+            jid = self.server.submit(j)
```

Comment:
You don't use jid.  It'll just get reset the next time through the loop.  I also don't see it used anywhere either.

Reply:
I use it outside the loop to get the start time of the last job that was submitted